prisma-fmt/src/get_config.rs

```diff
@@ -94,34 +101,132 @@ mod tests {
+            r#"{"config":{"generators":[],"datasources":[],"warnings":[]},"errors":[{"file_name":"schema.prisma","message":"\u001b[1;91merror\u001b[0m: \u001b[1mError validating: This line is invalid. It does not start with any known Prisma schema keyword.\u001b[0m\n  \u001b[1;94m-->\u001b[0m  \u001b[4mschema.prisma:5\u001b[0m\n\u001b[1;94m   | \u001b[0m\n\u001b[1;94m 4 | \u001b[0m\n\u001b[1;94m 5 | \u001b[0m            \u001b[1;91mdatasøurce yolo {\u001b[0m\n\u001b[1;94m 6 | \u001b[0m            }\n\u001b[1;94m   | \u001b[0m\n"},{"file_name":"schema.prisma","message":"\u001b[1;91merror\u001b[0m: \u001b[1mError validating: This line is invalid. It does not start with any known Prisma schema keyword.\u001b[0m\n  \u001b[1;94m-->\u001b[0m  \u001b[4mschema.prisma:6\u001b[0m\n\u001b[1;94m   | \u001b[0m\n\u001b[1;94m 5 | \u001b[0m            datasøurce yolo {\n\u001b[1;94m 6 | \u001b[0m            \u001b[1;91m}\u001b[0m\n\u001b[1;94m 7 | \u001b[0m        \n\u001b[1;94m   | \u001b[0m\n"},{"file_name":"schema.prisma","message":"\u001b[1;91merror\u001b[0m: \u001b[1mArgument \"provider\" is missing in generator block \"js\".\u001b[0m\n  \u001b[1;94m-->\u001b[0m  \u001b[4mschema.prisma:2\u001b[0m\n\u001b[1;94m   | \u001b[0m\n\u001b[1;94m 1 | \u001b[0m\n\u001b[1;94m 2 | \u001b[0m            \u001b[1;91mgenerator js {\u001b[0m\n\u001b[1;94m 3 | \u001b[0m            }\n\u001b[1;94m   | \u001b[0m\n"}]}"#
         ]];
+        let response = get_config(&request.to_string());
+        expected.assert_eq(&response);
+    }
+
+    #[test]
+    fn valid_generator_block() {
+        let schema = r#"
+            generator js {
+                provider = "prisma-client-js"
+                previewFeatures = ["prismaSchemaFolder"]
+            }
+        "#;
+
+        let request = json!({
+            "prismaSchema": schema,
+        });
 
-        let response = get_config(&request.to_string()).unwrap_err();
+        let expected = expect![[
+            r#"{"config":{"generators":[{"name":"js","provider":{"fromEnvVar":null,"value":"prisma-client-js"},"output":null,"config":{},"binaryTargets":[],"previewFeatures":["prismaSchemaFolder"]}],"datasources":[],"warnings":[]},"errors":[]}"#
+        ]];
+        let response = get_config(&request.to_string());
         expected.assert_eq(&response);
     }
 
     #[test]
-    fn get_config_missing_env_var() {
+    fn valid_generator_block_invalid_model() {
```

Comment:
This and the rest of `invalid` examples below is where old version would've completely stumpled and returned no config information.